docs/deployment/debian_ubuntu.md

```diff
@@ -0,0 +1,236 @@
+
+* `DEBIAN`: contains the `control` file.
+* `/usr/bin/`: contains the main executable and its starting script (recommended for starting your program via command-line).
+* `/usr/lib/your_program/`: where the `.so` native libraries will be included. (`your_program` is custom)
+* `/usr/share/applications/`: where the desktop shortcut will be placed.
+* `/usr/share/pixmaps/` and `/usr/share/icons/hicolor/**`: folders for application icons.
+
+:::info
+
+The `/usr/share/icons/hicolor/**` are *optional*, as in your app icon will probably show up on desktop even without those images, however, it is recommended to have them for better resolution.
+
+:::
+
+### 2) Make the `control` file
+
+The `control` file goes inside the `DEBIAN` folder.
+
+This file describes general aspects of your program, such as its name, version, category, maintainer, processor architecture and licenses. [Debian docs](https://www.debian.org/doc/debian-policy/ch-controlfields.html) have a more thorough description of all possible fields in the file.
+
+:::tip
+
+Author's comment: don't worry too much about filling all possible fields, most aren't required. This tutorial is to make a "good-enough" Debian package.
+
+:::
+
+Below is a simple example of a `control` file.
+
+```
+Package: your_program
+Version: 3.1.0
+Section: devel
+Priority: optional
+Architecture: amd64
+Installed-Size: 68279
+Maintainer: Ken Lee <kenlee@outlook.com>
+Homepage: https://github.com/kenlee/your_program
+Description: This is YourProgram, great for doing X.
+Copyright: 2022-2024 Ken Lee <kenlee@outlook.com>
+```
+
+### 3) Make the starter script
+
+This step is recommended for two reasons: first, to reduce the complexity of the desktop shortcut, and second, to make your app runnable from Terminal.
+
+The starter script file name should preferrably be `your_program` (without `.sh` extension), so whenever your user types "your_program" on the Terminal, he / she will start your program.
+
+To run from command-line, we need to append the libs directory path to the **LD_LIBRARY_PATH** environment variable - this is to make native libraries, such as `libHarfBuzzSharp.so` and `libSkiaSharp.so`, available to your program during initialization.
+
+:::warning
+
+Without this, your program will crash due to missing libraries!
+
+:::
+
+Example of `your_program` starter script, setting the variable in a command-scope:
+
+```sh
+#!/bin/bash
+LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib/your_program your_program_executable
+```
+
+### 4) Make the desktop shortcut
+
+The desktop shortcut file follows the [freedesktop specification](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#recognized-keys). Arch Linux Wiki also has [good related information](https://wiki.archlinux.org/title/Desktop_entries).
+
+Below is an example of a desktop shortcut file:
+
+```
+[Desktop Entry]
+Name=YourProgram
+Comment=YourProgram, great for doing X
+Icon=your_program
+Exec=your_program
```

Comment:
`Exec=your_program %F` if the app is supposed to open files (or `%U` if it's supposed to open URLs)

Reply:
Also need StartupWMClass entry that matches the app name or whatever is specified in X11PlatformOptions